lib/internal_metric/global.rb

```diff
@@ -368,10 +368,21 @@ def calc_postgres_highest_sequence
 
       RailsMultisite::ConnectionManagement.each_connection do |db|
         result[{ db: db }] = DB.query_single(<<~SQL)[0]
-          SELECT last_value
-          FROM pg_sequences
-          ORDER BY last_value DESC NULLS LAST
-          LIMIT 1
+          WITH columns_and_sequences AS (
+            SELECT table_name,
+                   column_name,
+                   information_schema.columns.data_type column_type,
+                   pg_sequences.data_type sequence_type,
+                   COALESCE(last_value, 0) last_value
+            FROM information_schema.columns
+            JOIN pg_sequences ON sequencename = REPLACE(REPLACE(column_default, 'nextval(''', ''), '''::regclass)', '')
+            WHERE information_schema.table_schema = 'public'
+          )
+          SELECT MAX(last_value)
+          FROM columns_and_sequences
+          WHERE column_type = 'integer' OR
+                -- The `id` column of these tables is a bigint, but the foreign key columns are usually integers
+                table_name IN ('reviewables', 'flags', 'sidebar_sections')
```

Comment:
It doesn't seem like this query works. I got this when I ran the query locally:

```
discourse_development=#           WITH columns_and_sequences AS (
            SELECT table_name,
                   column_name,
                   information_schema.columns.data_type column_type,
                   pg_sequences.data_type sequence_type,
                   COALESCE(last_value, 0) last_value
            FROM information_schema.columns
            JOIN pg_sequences ON sequencename = REPLACE(REPLACE(column_default, 'nextval(''', ''), '''::regclass)', '')
            WHERE information_schema.table_schema = 'public'
          )
          SELECT MAX(last_value)
          FROM columns_and_sequences
          WHERE column_type = 'integer' OR
                -- The `id` column of these tables is a bigint, but the foreign key columns are usually integers
                table_name IN ('reviewables', 'flags', 'sidebar_sections');
ERROR:  missing FROM-clause entry for table "information_schema"
LINE 9:             WHERE information_schema.table_schema = 'public'
                          ^
 ```

Reply:
Argh, this was a last second change that exploded.